src/hints/builtins.rs

```diff
@@ -95,23 +117,45 @@ pub fn update_builtin_ptrs(
     let builtin_ptrs = get_relocatable_from_var_name("builtin_ptrs", vm, ids_data, ap_tracking)?;
     let orig_builtin_ptrs = vm.get_relocatable(builtin_ptrs).unwrap();
     let selected_ptrs = get_relocatable_from_var_name("selected_ptrs", vm, ids_data, ap_tracking)?;
+    println!(" ***** update_builtin_ptrs()");
+    println!("    n_builtins: {}", n_builtins);
+    println!("    n_selected: {}", n_selected_builtins);
+    println!("    selected_encodings: {}", selected_encodings);
 
-    let all_builtins = vm.get_continuous_range(builtins_encoding_addr, n_builtins.deref().to_usize().unwrap()).unwrap();
-    let selected_builtins =
-        vm.get_continuous_range(selected_encodings, n_selected_builtins.deref().to_usize().unwrap()).unwrap();
+    // mimics the python `def update_builtin_pointers` but without explicitly passing in the args
+    let update_builtin_pointers = || -> Result<Vec<MaybeRelocatable>, HintError> {
+        let all_builtins = vm.get_continuous_range(builtins_encoding_addr, n_builtins.deref().to_usize().unwrap()).unwrap();
+        let selected_builtins =
+            vm.get_continuous_range(selected_encodings, n_selected_builtins.deref().to_usize().unwrap()).unwrap();
 
-    let mut returned_builtins: Vec<MaybeRelocatable> = Vec::new();
-    let mut selected_builtin_offset: usize = 0;
-    for (i, builtin) in all_builtins.iter().enumerate() {
-        if selected_builtins.contains(builtin) {
-            returned_builtins.push(vm.get_maybe(&(selected_ptrs + selected_builtin_offset)?).unwrap());
-            selected_builtin_offset += 1;
-        } else {
-            returned_builtins.push(vm.get_maybe(&(orig_builtin_ptrs + i)?).unwrap());
+        let mut returned_builtins: Vec<MaybeRelocatable> = Vec::new();
+        let mut selected_builtin_offset: usize = 0;
+        println!("    selected builtins: {:?}", selected_builtins);
+        for (i, builtin) in all_builtins.iter().enumerate() {
+            println!("    builtin: {}", builtin);
+            if selected_builtins.contains(builtin) {
+                println!("    - adding selected builtin {}", vm.get_maybe(&(selected_ptrs + selected_builtin_offset)?).unwrap());
+                returned_builtins.push(vm.get_maybe(&(selected_ptrs + selected_builtin_offset)?).unwrap());
+                selected_builtin_offset += 1;
+            } else {
+                let val = vm.get_maybe(&(orig_builtin_ptrs + i)?).unwrap();
+                println!("    - adding NON-selected builtin {}", val);
+                let val = if (val == MaybeRelocatable::RelocatableValue(Relocatable{segment_index: -3, offset: 0})) {
+                    MaybeRelocatable::RelocatableValue(Relocatable{segment_index: -3, offset: 1})
+                } else {
+                    val
+                };
```

Comment:
This ugly hack works around the `DiffAssertValues` issue by modifying a specific pointer when we encounter it.

Remove this in order to reproduce the issue; with it in place we are able to progress further through the OS code.